presto-parser/src/main/java/com/facebook/presto/sql/tree/ExpressionTreeRewriter.java

```diff
@@ -250,11 +252,57 @@ public Expression visitLogicalBinaryExpression(LogicalBinaryExpression node, Con
                 }
             }
 
-            Expression left = rewrite(node.getLeft(), context.get());
-            Expression right = rewrite(node.getRight(), context.get());
+            // It's common to have a large number of chained AND or OR. So we do it iteratively here to avoid stack overflow
```

Comment:
Wow, I'm curious how many chained AND/OR would trigger stack overflow...

Reply:
The failing test was testLargeQuerySuccess - AND of 500 1=1 - it fails in only raptortest - that too only sometimes apparently.

Reply:
BTW, in the other frontends I worked earlier, I always modeled AND/OR/+/UNION as variadic nodes for this reason as tools tend to generate long chains of these operators.